src/coord/src/coord.rs

```diff
@@ -2503,8 +2506,10 @@ impl Coordinator {
             }
         } else {
             // Use the earliest time that is still valid for all sources.
-            let (index_ids, _indexes_complete) = self.catalog.nearest_indexes(&[source_id]);
-            self.indexes.least_valid_since(index_ids)
+            let (index_ids, source_ids) = self.catalog.nearest_indexes(&[source_id]);
+            let mut since = self.indexes.least_valid_since(index_ids);
+            since.join_assign(&self.sources.least_valid_since(source_ids));
```

Comment:
New thought: I think this change does nothing? The sinces for the unmaterialized `source_ids` are all 0, so they will never cause `since` to advance, and `join_assign` never retreats.